build/common.sh

```diff
@@ -573,23 +542,47 @@ function kube::build::ensure_data_container() {
     # We want this to run as root to be able to chown, so non-root users can
     # later use the result as a data container.  This run both creates the data
     # container and chowns the GOPATH.
+    #
+    # The data container creates volumes for all of the directories that store
```

Comment:
Part of this change does away with the "tar up source and copy it over" and instead moves to a "bind mount source readonly".  This may slow down the virtualbox build.  I need to test that.  I don't remember if the slow part was read/write or just pure read path.  I'll test that out.

This change also makes sure that the /usr/local/go stuff is cached between runs.  That is critical for getting builds to be fast and incremental.